File: Algorithms/Medium/FindMinimumInRotatedSortedArray.py
# Time complexity: O(logn)
# Space complexity: O(1)
class Solution:
    def findMin(self, nums: list[int]) -> int:
        l, r = 0, len(nums) - 1
        while l < r:
            mid = (l + r) // 2
            if nums[mid] > nums[l]:
                l = mid + 1
            else:
                r = mid

        return nums[l]

File: Algorithms/Medium/FindMinimumInRotatedSortedArray.go
package medium

// Time complexity: O(logn)
// Space complexity: O(1)
func findMin(nums []int) int {
	l, r := 0, len(nums)-1
	for l < r {
		m := (l + r) / 2
		if nums[m] < nums[r] {
			r = m
		} else {
			l = m + 1
		}
	}

	return nums[l]
}
